homeassistant/components/anova_cooker/sensor.py

```diff
@@ -0,0 +1,85 @@
+"""Anova sous vide cooker integration."""
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.helpers.entity import Entity
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Add sensors for passed config_entry in HA."""
+    async_add_devices([AnovaCookerSensor(config_entry.data["device_id"])])
+
+
+class AnovaCookerSensor(Entity):
+    """Anova Cooker Sensor."""
+
+    def __init__(self, device_id):
+        """Set up the device_id, cooker, and empty state & attr."""
+        self._device_id = device_id
+        self._cooker = None
+        self._state = None
+        self._attr = {}
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "Anova Cooker"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    def update(self):
+        """Fetch new state data from the cooker"""
+        try:
+            if self._cooker is None:
+                from anova import AnovaCooker
+
+                self._cooker = AnovaCooker(self._device_id)
+            else:
+                self._cooker.update_state()
+        except:
```

Comment:
at the very least, use `except Exception as e` and then log the e so that users have some idea of why it is going wrong, ideally catch different types of errors and give detailed info about what went wrong.

Reply:
alas the python module return only an Exception, not different except classes.